TestBenches/FileReadUtility.h

```diff
@@ -177,82 +177,169 @@ unsigned int compareMemWithMem(const MemType& memory_ref, const MemType& memory,
-            std::cout << names[i] << "(" << std::dec << bit_widths[i] << ")" << spaces << "|";
-          }
-          if (j == 0) std::cout << "|";
-        }
+	// Print out column headers
+	int end_index = bit_widths.size();
+	if (start_index != 0)
+	  end_index = start_index + 1; // allows for only printing one column
+	
+	std::cout << " \t";
+	for (int j = 0; j < 2; j++) {
+	  for (int i = start_index; i < end_index; i++) {
+	    int length = bit_widths[i] - names[i].length() - 3;
+	    if (length < 0 ) length = 0;
+	    const std::string spaces(length,' ');
+	    std::cout << names[i] << "(" << std::dec << bit_widths[i] << ")" << spaces << "|";
+	  }
+	  if (j == 0) std::cout << "|";
+	}
       }
       std::cout << std::endl;
     }
 
     // If have reached the end of valid entries in both computed and reference, don't bother printing further
     if (data_com == 0 && data_ref == 0) continue;
-
+    
     // Print out the values
     std::cout << i << " \t";
     if (OutputBase == 2) std::cout << return_splitted(std::bitset<width>(data_ref).to_string(), bit_widths, names, start_index) << "|";
     else                 std::cout << std::hex << data_ref << "\t";
-
+    
     if (OutputBase == 2) std::cout << return_splitted(std::bitset<width>(data_com).to_string(), bit_widths, names, start_index) ;
     else                 std::cout << std::hex << data_com;
 
     // If there is extra entries in reference
     if (data_com == 0) {
       std::cout << "\t" << "<=== missing";
       if (!truncated) err_count++;
-    // If there is extra entries in computed
+      // If there is extra entries in computed
     } else if (data_ref == 0) {
       std::cout << "\t" << "<=== EXTRA";
       err_count++;
-    // If reference and computed entry are inconsistent
+      // If reference and computed entry are inconsistent
     } else if (data_com.to_long() != data_ref.to_long()) {
       std::cout << "\t" << "<=== INCONSISTENT";
       err_count++;
     }
-
     std::cout << std::endl;
   }
 
   return err_count;
 }
 
+template<class MemType, int OutputBase=2, int LSB=-1, int MSB=-1>
+unsigned int compareMemWithMemPage(const MemType& memory_ref, const MemType& memory,
+                                   const int ievt, const std::string& label,
+                                   const bool truncated = false,
+                                   const std::vector<int>& bit_widths = {}, const std::vector<std::string>& names = {},
+                                   const int start_index = 0)
+{
+  // This is a modification of compareMemWithMem to allow for outputting the
+  // data in binary format split by the various column types in the
+  // projections, AllProj.  Set start_index = 0 to see all columns, or any int
+  // between 1-5 to see only one at a time.
+
+  assert(bit_widths.size() == names.size());
+
+  unsigned int err_count = 0;
+
+  constexpr int width = (LSB >= 0 && MSB >= LSB) ? (MSB + 1) : MemType::getWidth();
```

Comment:
@aryd The width should take into account the LSB, in case it isn't zero.
```suggestion
  constexpr int width = (LSB >= 0 && MSB >= LSB) ? (MSB - LSB + 1) : MemType::getWidth();
```

Reply:
Restored this updated.